src/targets/gpu/kernels/include/migraphx/kernels/type_traits.hpp

```diff
@@ -26,20 +35,73 @@ struct enable_if<true, T>
 template <bool B, class T = void>
 using enable_if_t = typename enable_if<B, T>::type;
 
-template <class From, class To>
-struct is_convertible : bool_constant<__is_convertible(From, To)>
-{
-};
-
-template <class T, class U>
-struct is_same : false_type
-{
-};
-
-template <class T>
-struct is_same<T, T> : true_type
-{
-};
+// NOLINTNEXTLINE
+#define MIGRAPHX_BUILTIN_TYPE_TRAIT1(name)   \
+    template <class T>                       \
+    struct name : bool_constant<__##name(T)> \
+    {                                        \
+    }
+
+// NOLINTNEXTLINE
+#define MIGRAPHX_BUILTIN_TYPE_TRAIT2(name)      \
+    template <class T, class U>                 \
+    struct name : bool_constant<__##name(T, U)> \
+    {                                           \
+    }
+
+// NOLINTNEXTLINE
+#define MIGRAPHX_BUILTIN_TYPE_TRAITN(name)       \
+    template <class... Ts>                       \
+    struct name : bool_constant<__##name(Ts...)> \
+    {                                            \
+    }
+
+// MIGRAPHX_BUILTIN_TYPE_TRAIT1(is_arithmetic);
```

Comment:
Do we need these type traits commented out?

Reply:
Newer versions of clang do support these type traits so I would like to enable them in the future.